test/sql/filter/test_struct_pushdown.test

```diff
@@ -0,0 +1,126 @@
+# name: test/sql/filter/test_struct_pushdown.test
+# group: [filter]
+
+load __TEST_DIR__/test_struct_pushdown.test
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE integers(i STRUCT(a integer, b bool));
+
+statement ok
+INSERT INTO integers VALUES ({'a': 1, 'b': true}), ({'a': 2, 'b': false}), ({'a': 3, 'b': true}), ({'a': NULL, 'b': NULL});
```

Comment:
Can we add some more tests:

* Case insensitive struct extract (e.g. `i.A`)
* Top-level NULL value in `integers` (also it should probably not be called integers ;) )
* Nested struct extracts (when that is supported)
* String comparisons?